nvflare/fuel/utils/pipe/pipe_handler.py

```diff
@@ -109,7 +110,7 @@ def __init__(
         self.msg_cb = None
         self.msg_cb_args = None
         self.msg_cb_kwargs = None
-        self._other_end_is_up = threading.Event()
+        self.peer_is_up_or_dead = threading.Event()
 
```

Comment:
Need pause/resume methods for heartbeat mechanism.

Reply:
Added pause/resume function and tested with CyclicController